### DISCUSSION TITLE
How to manipulate StringIDs?

### DISCUSSION THREAD

**Discussion by TheMowgliMan** [OpenTTD/OpenTTD#12266] [Q&A]
I'm working on solving #10370, and I keep running into how difficult it is to manipulate the contents of a StringID. As far as I can tell, using StringIDs goes very low-level in the GUI code---far too far for an edit specifically for what I'm doing.

I want to truncate a StringID past the 32nd character in this case. What would likely be the best way to go about this?

**Comment by nielsmh** [comment]
I don't think you should limit how long NewGRF vehicle names can be, rather shouldn't the limit for player-made names be extended or removed?

**Reply by merni-ns** [reply]
Please see discussion in #12239  and #10384 ...

**Comment by merni-ns** [comment]
Apart from whether this is actually desirable to do...

You can't truncate a StringID, but you can truncate a string. Use `GetString` to get the `std::string` represented by a StringID. Then you can truncate it.

**Reply by TheMowgliMan** [reply]
All the GUI code requires a StringID to be passed; unless there is a way to create a StringID on the fly that I don't know about (likely), then that won't work most likely.
